### PR TITLE
[DOXIA-759] Support anchors in MarkdownSink

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java [hunk 1]
```diff
@@ -774,13 +778,30 @@ private void writeImage(String alt, String src) {
...
+            LOGGER.debug("{}Modified invalid anchor name '{}' to '{}'", getLocationLogPrefix(), name, id);
+        }
+
+        MutableAttributeSet att = new SinkEventAttributeSet();
+        att.addAttribute(SinkEventAttributes.ID, id);
+        att.addAttributes(atts);
+        StringBuilder htmlAnchor = new StringBuilder("<a");
+        htmlAnchor.append(SinkUtils.getAttributeString(att));
+        htmlAnchor.append(">");
+        htmlAnchor.append("</a>"); // close anchor tag immediately otherwise markdown would not be allowed afterwards
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:798]: Why not selfclosing element?

Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:798]: https://developer.mozilla.org/en-US/docs/Glossary/Void_element#self-closing_tags

Comment (michael-o) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownSink.java:798]: Interesting,  thanks.

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #756

**Pull request merged**